task_spec/spec/acceptance/init_spec.rb

```diff
@@ -50,8 +51,20 @@ def add_node(node, group_name, groups)
           config: config
         }
       else
-        # This is very unlikely to work
-        node_name, node = host.hostname
+        # host_hash is differnet depending on hypervisor (docker vs vmpooler)
+        # logic specific to vmpooler is contained here. I chose to copy pase
+        # some logic from above to keep this as a unit. Can refactor later.
+        config = { transport: 'ssh', ssh: { user: host.host_hash[:user] } }
+        key = nil
+        keys = host.connection.instance_variable_get(:@ssh).options[:keys]
+        key = keys.first if keys
+        config[:ssh][:"private-key"] = key if key
+        node_name = host.hostname
```

Comment:
We should probably still build node_name = "ssh://#{host.host_hash[:ip]}?n=#{host.hostname}"